app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningFragment.kt

```diff
@@ -94,18 +98,39 @@ class LearningFragment : Fragment() {
             }
             targetImageViews = updateTargetImageViews()
             learningPresenter =
-                LearningPresenter(dbMgt, args.learningMode, dataset, authPresenter, imageDisplayer)
+                LearningPresenter(
+                    dbMgt, args.learningMode, dataset, authPresenter, imageDisplayer,
+                    lifecycleScope
+                )
             learningPresenter.updateForNextSorting(
-                parentActivity,
                 targetImageViews,
                 binding.learningToSort
             )
             // sets the listeners for the image views of the sorting
             setEventListeners()
         }
+
+        visualFeedback = with(binding) {
+            LearningVisualFeedback(
+                lifecycleScope,
+                baseColor = getColor(requireContext(), R.color.white),
+                neutralColor = getColor(requireContext(), R.color.blue),
+                positiveColor = getColor(requireContext(), R.color.light_green),
+                negativeColor = getColor(requireContext(), R.color.red),
+                // TODO remove !! once enclosing CardViews put in all layouts
+                sourceCardView = learningToSortCv!!,
+                targetCardViews = listOf(
+                    learningCat0Cv!!,
+                    learningCat1Cv!!,
+                    learningCat2Cv!!
+                )
+            )
+        }.also { it.sourceCardViewShouldBlink(true) }
+
         requireActivity().onBackPressedDispatcher.addCallback(callback)
     }
 
+
```

Comment:
In the adaptDisplayToNumberOfCategories KDoc you have the wrong @param I think. (Sorry it's not from this PR but well..)

Reply:
thanks x)